src/richie/apps/core/context_processors.py

```diff
@@ -86,3 +92,90 @@ def site_metas(request):
 
     context["FRONTEND_CONTEXT"] = json.dumps(context["FRONTEND_CONTEXT"])
     return context
+
+
+class WebAnalyticsContextProcessor:
+    """
+    Context processor to add Web Analytics tracking information to Richie CMS templates and
+    frontend.
+    """
+
+    def context_processor(self, request: HttpRequest) -> dict:
+        """
+        Real implementation of the context processor for the Web Analytics core app sub-module
+        """
+        context = {}
+        if hasattr(request, "current_page"):
+            if getattr(settings, "WEB_ANALYTICS", None):
+                web_analytics_context = context["WEB_ANALYTICS"] = getattr(
+                    settings, "WEB_ANALYTICS", {}
+                )
+                if self.is_load_dimensions(request, web_analytics_context):
+                    web_analytics_context["DIMENSIONS"] = self.get_dimensions(request)
+        return context
+
+    # pylint: disable=no-self-use
+    def is_load_dimensions(self, request: HttpRequest, web_analytics_context) -> bool:
+        """
+        Method to easily enable or disable the load of web analytics dimensions
+        """
+        return "ENABLED" in web_analytics_context and web_analytics_context["ENABLED"]
+
```

Comment:
Through documentation, I understood that `WEB_ANALYTICS.ENABLED` aims to enable/disable the whole WEB_ANALYTICS part not just include or not dimensions?

Reply:
It should enable or disable the whole WEB_ANALYTICS. So when we have multiple web analytics software's we don't need to test if one of them is enabled.

Reply:
But if I understand well currently this flag is in charge to check if dimensions should be included within `web_analytics_context`.

Reply:
Used other approach suggested by @sampaccoud